src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -197,7 +201,10 @@ private void CheckAbstractClassImplementations(BindingDiagnosticBag diagnostics)
                     if (synthesizedImplementation.MethodImpl is { } methodImpl)
                     {
                         Debug.Assert(methodImpl is { Body: not null, Implemented: not null });
-                        methodImpls.Add(methodImpl);
+                        if (implementedMethods.Add(methodImpl.Implemented))
```

Comment:
I'm trying to understand the impact of choosing one of the interfaces (the first one) and dropping the other(s). 

When loading metadata produced by such a scenario, can we observe that we picked one interface implementation over the other? 
For instance, in terms of nullability warnings or symbol public APIs (`ExplicitInterfaceImplementations`)?

My initial reaction is that it doesn't feel right and we should instead merge the nullability differences between the two interfaces (possibly using `MergeEquivalentTypes` when we compute `AllInterfacesNoUseSiteDiagnostics`).

Looking for precedent, it looks like in this scenario we're already only emitting a single interface implementation (`implements` or `.interfaceimpl` in [IL](https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwARez3GDEMAriihMCgKYaUzlUCwAUM3m9gA4QBOCUEKDAAEATBgDCAHgAqAPgwgMASRlxW7DVkIkyFagBMoAZ3qV1m9tPMW820qZp091mznkBvDAF9mLi115+QVEJGXlFFT9XAmJ7PQxDE2cmaLxpNRTUnDtdKkdTKNc5DE8fFkzooQBmDFgESm4AMwgAY2oVaQBGOAxpEQ9vIA)). 
I looked at how are we determining which interface to emit (`InterfacesVisit`) and it seems we're also picking the first one. But that also seems odd to me.

Let's chat after standup to get others' perspectives.

Reply:
Some scenarios that would be interesting if we decide to merge instead of take the first one: 
- `I<object~>` vs `I<object!>` or `I<object?>`
- same scenarios but with variance: `I<out T>`
- scenarios with two type arguments: `I<object~, object!>` mixing with `I<object?, object~>` for instance

Reply:
Does the containing type's interface list also do some de-duplication when the same interface with different nullabilities is found?

If so, it seems like the interface which is used in the `.override` should just match whichever one is found in the containing type's interface list.

Reply:
From discussion with Aleksey, I'm fine to keep current design (select one interface rather than merging). That'd be a bigger change and is not required to fix the current bug